distribution/lib/Standard/Base/0.0.0-dev/src/Data/XML.enso

```diff
@@ -0,0 +1,385 @@
+
+    ## Get an attribute of an XML element.
+
+       Arguments:
+       - name: The name of the attribute to get.
+       - if_missing: The value returned if the attribute does not exist.
+
+       > Example
+         Get an attribute of an element.
+
+             root = XML_Document.from_string '<foo bar="one">hello</foo>' . root_element
+             root.attribute "bar"
+             # => "one"
+    attribute : Text -> Any -> Text | Any ! XML_Error
+    attribute self name:Text ~if_missing=(Error.throw (No_Such_Key.Error self name)) =
+        XML_Error.handle_java_exceptions <|
+            attr = self.java_element.getAttributeNode name
+            if attr.is_nothing then if_missing else attr.getValue
+
+    ## Gets a map containing f the attributes of an xml element.
+
+       > Example:
+           XML_Document.from_string '<foo bar="one">hello</foo>' . root_element . attributes
+           # => Map.from_vector [["bar", "one"]]
+    attributes : Map Text Text ! XML_Error
+    attributes self =
+        XML_Error.handle_java_exceptions <|
+            named_node_map = self.java_element.getAttributes
+            keys_and_values = 0.up_to named_node_map.getLength . map i->
+                node = named_node_map.item i
+                [node.getNodeName, node.getNodeValue]
+            Map.from_vector keys_and_values
+
+    ## Gets the text (non-markup) contents of the element and its descendants,
+       if any.
+
+       > Example
+         Get the text content of an element.
+
+             XML_Document.from_string '<foo>hello</foo>' . root_element . text
+             # => "hello"
+    text : Text ! XML_Error
+    text self =
+        XML_Error.handle_java_exceptions <|
+            self.java_element.getTextContent
+
+    ## Gets the raw XML of the element (including tag, attributes and contents).
+
+       > Example
+         Get the outer XML of an element.
+
+             XML_Document.from_string '<foo>hello</foo>' . root_element . outer_xml
+             # => '<foo>hello</foo>'
+    outer_xml : Text ! XML_Error
+    outer_xml self =
+        XML_Error.handle_java_exceptions <|
+            XML_Utils.outerXML self.java_element
+
+    ## Gets the raw XML of the contents of the element, not including the
+       outermost tag and attributes.
+
+       > Example
+         Get the inner XML of an element.
+
+             XML_Document.from_string '<foo><bar>hello</bar></foo>' . root_element . inner_xml
+             # => '<bar>hello</bar>'
+    inner_xml : Text ! XML_Error
+    inner_xml self =
+        XML_Error.handle_java_exceptions <|
+            XML_Utils.innerXML self.java_element
+
+    ## Gets elements matching a given tag name.
+
+       > Example
+             XML_Document.from_string '<foo> <baz>hello</baz> <baz>goodbye</baz> </foo>' . root_element . get_elements_by_tag_name "baz"
+             # => [XML_Document.from_string "<baz>hello</baz>" . root_element, XML_Document.from_string "<baz>goodbye</baz>" . root_element]
```

Comment:
I suggest to add within an example an element with a different tag name, to show it is skipped.

Reply:
Done.